aptos-move/replay-benchmark/src/overrides.rs

```diff
@@ -73,6 +125,114 @@ impl OverrideConfig {
+                };
+
+            let metadata = package.extract_metadata().unwrap_or_else(|err| {
+                panic!(
+                    "Failed to extract metadata for package {}: {:?}",
+                    package.name(),
+                    err
+                )
+            });
+            let new_package_state_value = old_package_state_value
+                .map_bytes(|bytes| {
+                    let mut package_registry = bcs::from_bytes::<PackageRegistry>(&bytes)
+                        .expect("Package registry should deserialize");
+
+                    let mut metadata_idx = None;
+                    for (idx, package_metadata) in package_registry.packages.iter().enumerate() {
+                        if package_metadata.name == metadata.name {
+                            metadata_idx = Some(idx);
+                            break;
+                        }
+                    }
+                    match metadata_idx {
+                        Some(idx) => {
+                            package_registry.packages[idx] = metadata;
+                        },
+                        None => {
+                            package_registry.packages.push(metadata);
+                        },
+                    }
+
+                    let bytes = bcs::to_bytes(&package_registry)
+                        .expect("Package registry should serialize");
+                    Ok(bytes.into())
+                })
+                .expect("Modifying package never returns an error");
+
+            overridden_package_registries
+                .insert(package_registry_state_key, new_package_state_value);
+
+            // Modify all existing modules or add new ones.
+            let bytecode_version = self.package_override.build_options.bytecode_version;
+            for module in package.modules() {
+                let mut module_bytes = vec![];
+                module
+                    .serialize_for_version(bytecode_version, &mut module_bytes)
+                    .unwrap_or_else(|err| {
+                        panic!(
+                            "Failed to serialize module {}::{}: {:?}",
+                            module.self_addr(),
+                            module.self_name(),
+                            err
+                        )
+                    });
+
+                let state_key = StateKey::module(module.self_addr(), module.self_name());
+                let onchain_state_value =
+                    state_view
+                        .get_state_value(&state_key)
+                        .unwrap_or_else(|err| {
+                            panic!(
+                                "Failed to fetch module {}::{}: {:?}",
+                                module.self_addr(),
+                                module.self_name(),
+                                err
+                            )
+                        });
+                let state_value = match onchain_state_value {
```

Comment:
I like map_or_else :D

Reply:
I think there is also an issue with the fact that you move `module_bytes` into closure, so you cannot use it for both default and map function? Nut I changed it anyway + added a clone, not a big deal here I guess